packages/smart-contracts/scripts-create2/xdeployer.ts

```diff
@@ -0,0 +1,163 @@
+  const wallets: Array<any> = [];
+  const signers: Array<any> = [];
+  const create2Deployer: Array<any> = [];
+  const createReceipt: Array<any> = [];
+  const result: Array<IDeploymentResult> = [];
+  let initcode: any;
+
+  const Contract = await hre.ethers.getContractFactory(contract);
+  if (constructorArgs && contract) {
+    initcode = await Contract.getDeployTransaction(...constructorArgs);
+  } else if (!constructorArgs && contract) {
+    initcode = await Contract.getDeployTransaction();
+  }
+
+  for (let i = 0; i < hre.config.xdeploy.rpcUrls.length; i++) {
+    providers[i] = new hre.ethers.providers.JsonRpcProvider(hre.config.xdeploy.rpcUrls[i]);
+    if (hre.config.xdeploy.networks[i] === 'celo') {
+      const originalBlockFormatter = providers[i].formatter._block;
+      providers[i].formatter._block = (value: any, format: any) => {
+        return originalBlockFormatter(
+          {
+            gasLimit: constants.Zero,
+            ...value,
+          },
+          format,
+        );
+      };
+    }
+    wallets[i] = new hre.ethers.Wallet(hre.config.xdeploy.signer, providers[i]);
+    signers[i] = wallets[i].connect(providers[i]);
+
+    let computedContractAddress: string;
+    if (
+      hre.config.xdeploy.networks[i] !== 'hardhat' &&
+      hre.config.xdeploy.networks[i] !== 'localhost'
+    ) {
+      create2Deployer[i] = new hre.ethers.Contract(
+        hre.config.xdeploy.deployerAddress,
+        requestDeployer.getContractAbi(),
+        signers[i],
+      );
+      try {
+        computedContractAddress = await create2Deployer[i].computeAddress(
+          hre.ethers.utils.id(hre.config.xdeploy.salt),
+          hre.ethers.utils.keccak256(initcode.data),
+        );
+      } catch (err) {
+        throw new Error(
+          'Contract address could not be computed, check your contract name and arguments',
+        );
+      }
+      try {
+        createReceipt[i] = await create2Deployer[i].deploy(
+          AMOUNT,
+          hre.ethers.utils.id(hre.config.xdeploy.salt),
+          initcode.data,
+          { gasLimit: hre.config.xdeploy.gasLimit },
+        );
+
+        createReceipt[i] = await createReceipt[i].wait();
+
+        result[i] = {
+          network: hre.config.xdeploy.networks[i],
+          contract: contract,
+          address: computedContractAddress,
+          receipt: createReceipt[i],
+          deployed: true,
+          error: undefined,
+        };
+      } catch (err) {
+        result[i] = {
+          network: hre.config.xdeploy.networks[i],
+          contract: contract,
+          address: computedContractAddress,
+          receipt: undefined,
+          deployed: false,
+          error: err,
+        };
+      }
+    } else if (
+      hre.config.xdeploy.networks[i] === 'hardhat' ||
+      hre.config.xdeploy.networks[i] === 'localhost'
+    ) {
+      console.log('ici');
+      const hhcreate2Deployer = await hre.ethers.getContractFactory('Create2DeployerLocal');
```

Comment:
Does it work? Shouldn't it be: 
```suggestion
      const hhcreate2Deployer = await hre.ethers.getContractFactory('RequestDeployer');
```
?